f2/src/ThankYouPage.js

```diff
@@ -54,17 +54,23 @@ export const ThankYouPage = () => {
             <H1 mb={6}>
               <Trans id="thankYouPage.title" />
             </H1>
-            <P fontSize="lg">
-              <Trans
-                id="thankYouPage.referenceNumber"
-                values={{
-                  reference: state.formData.reportId
-                    ? state.formData.reportId
-                    : '< report ID >',
-                }}
-              >
-                <Text as="span" d="block" fontSize="2xl" />
-              </Trans>
+
+            <P fontSize="2xl">
+              {state.formData.reportId &&
+              state.formData.reportId.startsWith('NCFRS-') ? (
+                <Trans id="thankYouPage.reportSubmissionError">
+                  <Text as="span" d="block">
+                    {state.formData.reportId}
+                  </Text>
+                </Trans>
+              ) : (
+                <Trans
+                  id="thankYouPage.referenceNumber"
+                  values={{ reference: state.formData.reportId }}
+                >
+                  <Text as="span" d="block" />
+                </Trans>
+              )}
```

Comment:
* We should flip the order of these two blocks (ie show the reportId when `state.formData.reportId && state.formData.reportId.startsWith('NCFRS-')`)
* also, we should have a third option... something like when the confirmation page loads (ie in `ConfirmationPage.js` after the `const { i18n } = useLingui()`)
```
if formdata.reportId !== '') 
 dispatch({ type: 'saveFormData', data: { reportId: '' } })
```
 to make the reportId an empty string. Then on the thankyou page we could also check for `reportId ===''` and if so show something like "Submitting report..." (which would then get replaced by the number or the error message when there's a response from the server)

Reply:
Sounds good! I'm on it

Reply:
@sastels fixed! Hopefully I covered all the cases :)